core/templates/components/summary-tile/story-summary-tile.component.ts

```diff
@@ -32,25 +32,29 @@ import { StorySummary } from 'domain/story/story-summary.model';
   templateUrl: 'story-summary-tile.component.html'
 })
 export class StorySummaryTileComponent implements OnInit {
-  @Input() classroomUrlFragment: string;
-  @Input() storySummary: StorySummary;
-  @Input() topicUrlFragment: string;
-  initialCount: number;
-  chaptersDisplayed: number;
-  nodeCount: number;
-  completedStoriesCount: number;
-  storyProgress: number;
-  thumbnailUrl: string;
-  showButton: boolean;
+  // These properties are initialized using Angular lifecycle hooks
+  // and component interactions, therefore we need to do non-null assertion,
+  // for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() classroomUrlFragment!: string;
+  @Input() storySummary!: StorySummary;
+  @Input() topicUrlFragment!: string;
+  initialCount!: number;
+  chaptersDisplayed!: number;
+  nodeCount!: number;
+  completedStoriesCount!: number;
+  storyProgress!: number;
+  thumbnailUrl!: string | null;
+  storyLink!: string | null;
```

Comment:
If this can be null, why not just assign it null and remove the `!`

Reply:
done